BDArmory/Control/BDModulePilotAI.cs

```diff
@@ -1930,7 +1937,7 @@ protected override void AutoPilot(FlightCtrlState s)
             CalculateAccelerationAndTurningCircle();
             CheckFlatSpin();
 
-            if ((float)vessel.radarAltitude < minAltitude)
+            if ((float)vessel.radarAltitude < minAltitude && !isBombing)
```

Comment:
Testing this, its removal breaks torpedo bombing and causes the craft to go 'Engaging' then immediately go 'Gain Alt' as soon as the alt goes below minAlt, if minAlt is higher than 200, for both hard and soft minAlt

Reply:
Having the Hard Min Alt toggle enabled does result in correct behavior and no interruptions from 'Gain Alt'.

Reply:
> Testing this, its removal breaks torpedo bombing and causes the craft to go 'Engaging' then immediately go 'Gain Alt' as soon as the alt goes below minAlt, if minAlt is higher than 200, before returning to 'Engaging' and diving down to torp alt again, triggering 'Gain Alt' again...

Does torpedo bombing set some status other than those allowed by `gainAltInhibited`? If so, that would cause this. I still think the better way is not to avoid setting `belowMinAltitude` (as that is also used for other logic elsewhere, such as in `FlyToPosition`) but to include the necessary conditions for torp/bombing in the `gainAltInhibited` logic.

> Having the Hard Min Alt toggle enabled does result in correct behavior and no interruptions from 'Gain Alt'.

That sounds backwards. Having Hard Min Alt enabled prevents `gainAltInhibited` from being set, meaning that being below min alt will trigger the Gain Alt behaviour. I'll try to test this tomorrow.

Reply:
No. The only thing that gets set is the expected ``isBombing`` bool; the only difference is Torp bombing sets``finalBombingAlt`` to 200 instead of ``bombingAltitude``. The Aeris I was using for testing had a minAlt set to 450m.

Reply:
Testing with
```diff
diff --git a/BDArmory/Control/BDModulePilotAI.cs b/BDArmory/Control/BDModulePilotAI.cs
index b2e2d97c5..4ba1f3280 100644
--- a/BDArmory/Control/BDModulePilotAI.cs
+++ b/BDArmory/Control/BDModulePilotAI.cs
@@ -1939,7 +1939,7 @@ protected override void AutoPilot(FlightCtrlState s)
 
-            if ((float)vessel.radarAltitude < minAltitude && !isBombing) //TODO - refinement of torp bombing temporary minAlt ignore needed
+            if ((float)vessel.radarAltitude < minAltitude)// && !isBombing) //TODO - refinement of torp bombing temporary minAlt ignore needed
             { belowMinAltitude = true; }
 
-            if (gainAltInhibited && (!belowMinAltitude || !(currentStatusMode == StatusMode.Engaging || currentStatusMode == StatusMode.Evading || currentStatusMode == StatusMode.RammingSpeed || currentStatusMode == StatusMode.GainingAltitude)))
-            { // Allow switching between "Engaging", "Evading", "Ramming speed!" and "Gain Alt." while below minimum altitude without disabling the gain altitude inhibitor.
+            if (gainAltInhibited && (!belowMinAltitude || !(isBombing || currentStatusMode == StatusMode.Engaging || currentStatusMode == StatusMode.Evading || currentStatusMode == StatusMode.RammingSpeed || currentStatusMode == StatusMode.GainingAltitude)))
+            { // Allow switching between "Engaging", "Bombing", "Evading", "Ramming speed!" and "Gain Alt." while below minimum altitude without disabling the gain altitude inhibitor.
                 gainAltInhibited = false;
@@ -1948,4 +1948,4 @@ protected override void AutoPilot(FlightCtrlState s)
 
-            if (!hardMinAltitude && !gainAltInhibited && belowMinAltitude && (currentStatusMode == StatusMode.Engaging || currentStatusMode == StatusMode.Evading || currentStatusMode == StatusMode.RammingSpeed) && !vessel.InNearVacuum())
-            { // Vessel went below minimum altitude while "Engaging", "Evading" or "Ramming speed!", enable the gain altitude inhibitor.
+            if (!hardMinAltitude && !gainAltInhibited && belowMinAltitude && (isBombing || currentStatusMode == StatusMode.Engaging || currentStatusMode == StatusMode.Evading || currentStatusMode == StatusMode.RammingSpeed) && !vessel.InNearVacuum())
+            { // Vessel went below minimum altitude while "Engaging", "Bombing", "Evading" or "Ramming speed!", enable the gain altitude inhibitor.
                 gainAltInhibited = true;
```
and using 200 bombing alt, 500 min alt and 600 default alt on the quicksave I shared earlier, it was behaving as I'd expect, that is, inhibiting gain alt behaviour while extending or approaching on the bombing run (except briefly at the end of the extending where it was briefly not considered bombing, so something would need to be tweaked there).
Also, using hard min alt prevents it from going below min even when bombing.

So, if it's not doing that on your end, then there must be some other change that's interfering somewhere.

Reply:
Testing with the above changes has been working so far in testing on my end